android-sdk/src/androidTest/java/com/mobilecoin/lib/TxOutStoreTest.java

```diff
@@ -149,4 +167,140 @@ public void fetch_fog_misses_test()
         }
     }
 
+    @Test
+    public void testEmptyResponse() throws Exception {
+        AttestedViewClient viewClient = mock(AttestedViewClient.class);
+        View.QueryResponse.Builder responseBuilder = View.QueryResponse.newBuilder();
+        when(viewClient.request(any(), eq(Long.valueOf(0L)), eq(Long.valueOf(0L)))).thenReturn(responseBuilder.build());
+
+        TxOutStore uut = new TxOutStore(null);
+        Set<BlockRange> results = uut.updateRNGsAndTxOuts(viewClient,
+                new DefaultFogQueryScalingStrategy(), new DefaultFogSeedProvider(),
+                new DefaultVersionedCryptoBox());
+        assertTrue(results.isEmpty());
+    }
+
+    @Test
+    public void testUpdateRNGsAndTxOuts() throws Exception {
+
+        //build AccountKey
+        AccountKey accountKey = mock(AccountKey.class);
+
+        //build Kex RNG pub key
+        KexRng.KexRngPubkey.Builder kexRngPubkey = KexRng.KexRngPubkey.newBuilder();
+        kexRngPubkey.setVersion(0);
+        kexRngPubkey.setPubkey(ByteString.copyFrom(new byte[32]));
+
+        //build RNG record
+        View.RngRecord.Builder rngRecord = View.RngRecord.newBuilder();
+        rngRecord.setStartBlock(0L);
+        rngRecord.setIngestInvocationId(0L);
+        rngRecord.setPubkey(kexRngPubkey.build());//add Kex RNG pub key;
+
+        //build TxOutStoreResult
+        View.TxOutSearchResult.Builder searchResult = View.TxOutSearchResult.newBuilder();
+        searchResult.setResultCode(View.TxOutSearchResultCode.NotFound_VALUE);
+        searchResult.setCiphertext(ByteString.copyFrom(new byte[32]));
+        searchResult.setSearchKey(ByteString.copyFrom(new byte[32]));
+
+        //build query responses
+        View.QueryResponse.Builder responseBuilder = View.QueryResponse.newBuilder();
+        responseBuilder.addRngs(rngRecord.build());//add RNG record
+        View.QueryResponse response1 = responseBuilder.build();
+        responseBuilder.addTxOutSearchResults(searchResult.build());
+        View.QueryResponse response2 = responseBuilder.build();
+
+        //build FogSeed
+        FogSeed fogSeed = mock(FogSeed.class);
+        when(fogSeed.isObsolete()).thenReturn(false);
+        when(fogSeed.getNextN(anyLong())).thenReturn(new byte[1][32]);
+        when(fogSeed.getOutput()).thenReturn(new byte[32]);
+
+        //build FogSeedProvider
+        FogSeedProvider seedProvider = mock(FogSeedProvider.class);
+        when(seedProvider.fogSeedFor(any(), any())).thenReturn(fogSeed);
+
+        //build VersionedCryptoBox
+        VersionedCryptoBox cryptoBox = mock(VersionedCryptoBox.class);
+        when(cryptoBox.versionedCryptoBoxDecrypt(any(), any())).thenReturn(new byte[] {17, -93, 2, -81, 7, -62,
```

Comment:
That's a good idea. It'll improve readability because this byte array looks like nonsense and takes up a lot of space. This byte array is just an instance of a bytestring that I copied from a sample TxOutRecord that just needs to be there so it parses the object correctly. I will move this to the bottom of the file and make it a private static final field.